### PR TITLE
Hellhound gas nerf

### PR DESCRIPTION
**Pull request by Couls**
**What does this PR do:**
Removes the hellhound smoke ability only limiting it to greater hellhounds
**Changelog:**
:cl:
del: Hellhound smoke has been removed from hellhounds
/:cl:

### CODE REVIEW

**Comment by KasparoVy** [comment]
Make a subtype called 'heckhound' and neuter it of its AOE :rofl:

**Comment by Couls** [comment]
> Make a subtype called 'heckhound' and neuter it of its AOE 🤣

yes. yes. YES!

**Comment by Kyep** [comment]
Simply wearing internals makes you immune to this attack.
Everyone starts with internals.
It is very easy to counter this attack.

**Comment by Mitchs98** [comment]
Something I noticed that is very buggy with both this and the Psychocomps gas...NOBREATHE races are somehow affected by it. Even IPC have to wear internals last I checked? :thinking:

**Comment by Couls** [comment]
> Simply wearing internals makes you immune to this attack.
> Everyone starts with internals.
> It is very easy to counter this attack.

I was told this has a chance of getting through internals, a few races wouldn't normally be wearing internals on station and its really effective crowd-control to deal with security. I still think this attack is really powerful because it takes preparation for the victim while the hellhound has it outright and doesn't even need to activate it as an ability. I also would like to add that I didn't know about internals being a counter and a lot of other people would probably be included in this. I think the heal and the high melee damage is already really strong traits for hellhounds

**Comment by Kyep** [comment]
I just tested it, and even as a standard civilian I was immune to it when I had internals active.

**Comment by Couls** [comment]
I feel like that’s a good compromise though the only issue is that they already have that tag so I guess there’s a bug in the flag for both nobreathe and the hostile spawns

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -1070,10 +1070,10 @@ so that different stomachs can handle things in different ways VB*/
 /mob/living/carbon/proc/can_breathe_gas()
 	if(!wear_mask)
 		return TRUE
-
 	if(!(wear_mask.flags & BLOCK_GAS_SMOKE_EFFECT) && internal == null)
-		return TRUE
-
+                return TRUE
+        if((BREATHLESS in H.mutations) || (NO_BREATHE in H.dna.species.species_traits)
+	        return TRUE
```
Comment (Kyep) [inline code/modules/mob/living/carbon/carbon.dm]: Shouldn't this be above, and FALSE?

File: code/modules/mob/living/simple_animal/hostile/hellhound.dm [hunk 2]
```diff
@@ -95,7 +95,7 @@
 		return TRUE
 	return FALSE
 
-/mob/living/simple_animal/hostile/hellhound/AttackingTarget()
+/mob/living/simple_animal/hostile/hellhound/greater/AttackingTarget()
```
Comment (Kyep) [inline code/modules/mob/living/simple_animal/hostile/hellhound.dm]: This change needs to be taken out.

**Comment by Kyep** [comment]
@Couls You need to test those logic changes in carbon.dm. I worry it will evaluate "if(!(wear_mask.flags & BLOCK_GAS_SMOKE_EFFECT) && internal == null)" and return TRUE before it even gets to the "if((BREATHLESS IN H.mutations"...  check and thus never realize it should have returned false due to that.

**Comment by Couls** [comment]
Yeah I know that’s why I set it to Do not merge, I need to make sure everything compiles and runs properly and I also want to look into why hostile mobs don’t spawn hostile if you use blood in gold cores

File: code/modules/mob/living/carbon/carbon.dm [hunk 3]
```diff
@@ -1070,10 +1070,10 @@ so that different stomachs can handle things in different ways VB*/
 /mob/living/carbon/proc/can_breathe_gas()
 	if(!wear_mask)
 		return TRUE
-
 	if(!(wear_mask.flags & BLOCK_GAS_SMOKE_EFFECT) && internal == null)
-		return TRUE
-
+                return TRUE
+        if((BREATHLESS in H.mutations) || (NO_BREATHE in H.dna.species.species_traits)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/carbon.dm]: We're not making some races immune to chem smoke. Not now, not ever.

**Comment by Fox-McCloud** [comment]
This impacts way more than just hellhounds. This is making a decent amount of races out and out immune to chem smoke, regardless of the situation. Sorry, no.

IF hellhounds are this problematic for their smoke, them @KasparoVy's suggestion is best; make a subtype that doesn't spawn smoke. Make it so that the ones that have smoke can't be made by xenobio, but the ones that don't have smoke can be made there.

I suspect that's ultimately what you're trying to address here anyway.

**Comment by Couls** [comment]
Yeah that was the initial idea, I guess I can make a heckhound and then revert the changes making nobreath races immune to smoke

**Comment by Couls** [comment]
Right so i'll just make two PRs, one to make a subtype of hellhound called heckhound without the smoke AOE and then one PR about making xenobio more risk for the reward

**Comment by Couls** [comment]
ALRIGHT, after talking to kyet we've come to the conclusion that we're just gonna remove the smoke from normal hellhounds instead of making a heckhound subtype

**Comment by Kyep** [comment]
Most people I've talked to about this seem to be against changing hellhounds at all.
But I'll leave it to them to comment.

**Comment by Exavere** [comment]
After looking at their Health, Low/High Damage Ratios, without the smoke they'd essentially just be another Araneus.  I also agree on the fact that all you have to do is wear internals to counter the Chemicals, which you carry around at round start.  They're meant to be dangerous, not just another simple mob that doesn't have any presence.  I've never had an instance where a Hell-hound pops up and I go "Oh god, it's OP we're all doomed", at least for me personally.

**Comment by Couls** [comment]
> After looking at their Health, Low/High Damage Ratios, without the smoke they'd essentially just be another Araneus. I also agree on the fact that all you have to do is wear internals to counter the Chemicals, which you carry around at round start. They're meant to be dangerous, not just another simple mob that doesn't have any presence. I've never had an instance where a Hell-hound pops up and I go "Oh god, it's OP we're all doomed", at least for me personally.

You do have to keep in mind you play a race that constantly has internals so their strongest attack on you is instantly countered

**Comment by Imotepchief** [comment]
Hellhound smoke isn't really that big of an issue for people once they are aware of the counter, which is simple to achieve for every single player, even if they somehow lost their original internals box, there are others scattered around the station.
I don't think the smoke is sufficiently powerful to nerf at all to be honest and to be fair, Hellhounds are outclassed in terms of combat potential when faced with like a xenoqueen.
If the smoke were to be removed, there would be no technical reason to favor getting a hellhound over a xenoqueen, as a xenoqueen is essentially a hellhound with a ranged attack instead of smoke.

The real question here is, what exactly prompted this particular desire to remove smoke, as from my perspective, having fought, been and having hellhounds as pets, I don't find them overpowered of worthy of a nerf.

**Comment by Couls** [comment]
Honestly a lone hellhound isn’t ALL that overpowered but nothing is stopping xenobio from mass producing them I feel like this is a topic that needs a lot more discussion and hellhounds aren’t the only thing that needs looking into, I think I raised a bit of awareness on the situation which is definitely helpful and you are definitely right about xeno queen being the next mass produce target if hellhound is nerfed though a lot would still use hellhound for its ability to heal itself even without the smoke

**Comment by Pixelzilla** [comment]
Hellhounds are not a common spawn from xenobio in the first place. It takes a lot of gold slime extracts and RNG luck to get a single hellhound and even then the only way a hellhound is going to be used in combat is by an antag or in extreme situations. You also have to consider that you need ghosts willing to play as a hellhound if you're going sentience potion route OR you need someone dedicating their time to rainbow slimes and giving up their normal body to become a hellhound.

I dont think hellhounds are OP or need nerfs. I think people just need to accept the fact that you cant just toolbox everything you see and you should be more aware of your surroundings and maybe choose to flee and not engage in a fight when you're not likely to win. There is nothing wrong with xeno queens either. If anything xenobiology mobs in general need more functionality to them to make them more useful and unique like the hellhound is.

**Comment by Couls** [comment]
Both the mobs you stated are actually much harder to flee than they are to fight, the alien queen can use its spit if you decide to leg it down the hallway and if you get hit by the hellhound smoke without internals on.. well you're not going anywhere. not to mention most ghosts won't say no to being a hellhound because of the opportunity to murderbone and it kind of shows(They're usually really quick to jump the gun in tense situations) not to add their ability to understand every language and give a demonic whisper which is both fun for rp and unique to hellhounds

**Comment by variableundefined** [comment]
@Couls Is this PR ready for merge? What's the status on this?

**Comment by Couls** [comment]
@variableundefined Sorry forgot to update the status but the PR is ready to merge

**Comment by Couls** [comment]
Don’t think this PR is necessary anymore since new crit has been merged and people can take much more damage

**Pull request closed without merging**